server/src/main/java/org/opensearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -421,6 +499,63 @@ public InternalAggregation buildEmptyAggregation() {
         return rangeFactory.create(name, buckets, format, keyed, metadata());
     }
 
+    class MatchedRange {
```

Comment:
This class can be `static`, since it doesn't use anything from the enclosing scope.